torch_geometric/nn/conv/gatv2_conv.py

```diff
@@ -186,23 +185,22 @@ def __init__(
         else:
             self.lin_edge = None
 
-        # The number of out channels for residual and bias depending on concat
-        if concat:
-            out_chennels_total = out_channels * heads
-        else:
-            out_chennels_total = out_channels
+        # The number of output channels:
+        total_out_channels = out_channels * (heads if concat else 1)
 
         if residual:
-            if in_channels != out_chennels_total:
-                self.lin_res = Linear(in_channels, out_channels * heads,
-                                      bias=bias, weight_initializer='glorot')
-            else:
-                self.lin_res = Identity()
+            self.res = Linear(
+                in_channels
+                if isinstance(in_channels, int) else in_channels[1],
+                total_out_channels,
+                bias=False,
+                weight_initializer='glorot',
+            )
```

Comment:
In this way, a linear transformation will also be applied when `in_channels == total_out_channels`. It is consistent with the original `example/ppi.py` but over-complicates the idea of skip-connection.

Maybe it's better to directly add the residual without linear transformation following the GAT authors' official implementation  here:
> https://github.com/PetarV-/GAT/blob/5af87e7fce2b90ae1cbd621cd58059036a3c7436/utils/layers.py#L27-L32